flink-streaming-connectors/flink-connector-nifi/src/main/java/org/apache/flink/streaming/connectors/nifi/NiFiSource.java

```diff
@@ -143,4 +144,13 @@ public void close() throws Exception {
 		client.close();
 	}
 
+ /**
+	* {@inheritDoc}
+	* <p>
+	* Sets the {@link #isRunning} flag to {@code false}.
+	*/
+	@Override
+	public void stop() {
+		this.isRunning = false;
```

Comment:
I'd like other to chime in here: do you have to call close() within stop()? The Twitter source is doing this.